internal/controller/api/v1alpha1/cluster_types.go

```diff
@@ -25,17 +25,86 @@ import (
 
 // ClusterSpec defines the desired state of Cluster
 type ClusterSpec struct {
```

Comment:
some fields here have `omitempty`, but seem mandatory: e.g base_dns_domain, cluster_network_cidr, etc

Reply:
I was thinking the same but they are not required for the creation, user can create a cluster without defining those fields, he will not be able to install it the same as we have in out rest API.